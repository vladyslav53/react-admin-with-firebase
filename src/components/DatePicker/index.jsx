import React from 'react';
import DatePicker, { registerLocale } from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';
import PropTypes from 'prop-types';
import es from 'date-fns/locale/es';
import en from 'date-fns/locale/en-US';

import './DatePicker.scss';

const DatePickerStyled = ({ name, date, setState }) => {
  const onDateChangedHandler = value =>
    setState(prevState => ({
      ...prevState,
      [name]: value.toDateString()
    }));

  registerLocale('en', en);
  registerLocale('es', es);

  const local = navigator.language.split(/[-_]/)[0];

  const dateFormat = navigator.language === 'es-US' ? 'MM-dd-yy' : 'dd-MM-yy';

  return (
    <DatePicker
      locale={local}
      dateFormat={dateFormat}
      selected={date}
      onChange={onDateChangedHandler}
    />
  );
};

DatePickerStyled.propTypes = {
  name: PropTypes.string.isRequired,
  date: PropTypes.instanceOf(Date).isRequired,
  setState: PropTypes.func.isRequired
};

export default DatePickerStyled;
