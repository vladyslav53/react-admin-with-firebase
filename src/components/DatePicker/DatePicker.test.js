import React from 'react';
import { shallow, mount } from 'enzyme';
import DatePicker from 'react-datepicker';

import DatePickerStyled from '.';

describe('<DatePickerStyled /> rendering', () => {
  const setState = jest.fn();

  it('should render without crashing', () => {
    const component = shallow(
      <DatePickerStyled name="test" date={new Date('Thu Nov 12 2020 00:00:00 GMT-0000')} setState={setState} />
    );

    expect(component).toMatchSnapshot();
  });

  it('should render <DatePicker /> component correctly', () => {
    const component = mount(
      <DatePickerStyled name="test" date={new Date('11/12/2020')} setState={setState} />
    );

    expect(component.exists(DatePicker)).toBeTruthy();
  });

  it('should pass the date prop to <DatePicker selected={date} /> correctly', () => {
    const component = mount(
      <DatePickerStyled name="test" date={new Date('11/12/2020')} setState={setState} />
    );

    expect(component.find(DatePicker).prop('selected')).toEqual(
      new Date('11/12/2020')
    );
  });
});
