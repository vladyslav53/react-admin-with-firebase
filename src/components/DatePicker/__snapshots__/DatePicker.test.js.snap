// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePickerStyled /> rendering should render without crashing 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <DatePickerStyled
      date={2020-11-12T00:00:00.000Z}
      dateFormat="en-US"
      name="test"
      setState={[MockFunction]}
    />
  </Provider>
</Router>
`;
